src/scripts/main.js

```diff
@@ -1,3 +1,226 @@
+  const rows = [...document.querySelectorAll('tbody tr')];
+  const headClicked = e.target.textContent;
+  const indexHead = headings.indexOf(headClicked);
+
+  if (prevClicked !== headClicked) {
+    clickCounter = 0;
+  }
+
+  clickCounter++;
+
+  const sortedRow = rows.sort((elem1, elem2) => {
+    const value1 = elem1.children[indexHead].textContent;
+    const value2 = elem2.children[indexHead].textContent;
+    const howTowSort = clickCounter % 2 !== 0;
+
+    if (indexHead === 3 || indexHead === 4) {
+      const num1 = indexHead === 4 ? +value1.replace(/[^0-9]+/g, '') : +value1;
+      const num2 = indexHead === 4 ? +value2.replace(/[^0-9]+/g, '') : +value2;
+
+      return howTowSort ? num1 - num2 : num2 - num1;
+    }
+
+    return howTowSort
+      ? value1.localeCompare(value2)
+      : value2.localeCompare(value1);
+  });
+
+  actualBody.innerHTML = '';
+
+  sortedRow.forEach((row) => {
+    actualBody.append(row);
+  });
+  prevClicked = headClicked;
+});
+
+tbody.addEventListener('click', (e) => {
+  const selectedElem = e.target;
+  const rows = [...document.querySelectorAll('tbody tr')];
+
+  if (prevRow !== undefined) {
+    prevRow.classList.remove('active');
+  }
+
+  const row = rows.find((elem) => {
+    const childrenOfRow = [...elem.children];
+
+    return childrenOfRow[childrenOfRow.indexOf(selectedElem)];
+  });
+
+  row.classList.toggle('active');
+  prevRow = row;
+});
+
+tbody.addEventListener('dblclick', (e) => {
+  const selectedElem = e.target;
+  const oldText = e.target.textContent;
+  const newValue = document.createElement('input');
+
+  newValue.classList.add('cell-input');
+  newValue.value = selectedElem.textContent;
+
+  const row = selectedElem.closest('tr');
+
+  if (row) {
+    selectedElem.textContent = '';
+    selectedElem.appendChild(newValue);
+    newValue.focus();
+
+    ['blur', 'keydown'].forEach((value) => {
+      newValue.addEventListener(value, (keyPress) => {
+        if (
+          value === 'blur' ||
+          (value === 'keydown' && keyPress.key === 'Enter')
+        ) {
+          selectedElem.textContent =
+            newValue.value === '' ? oldText : newValue.value;
+        }
+      });
+    });
+  }
+});
+// #endregion
+
+const buttonAdd = document.querySelector('form button');
+
+buttonAdd.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  let isValid = false;
+  const valuesToAdd = [];
+  const elemForm = document.querySelectorAll('[data-qa]');
+
+  elemForm.forEach((elem) => {
+    valuesToAdd[valuesToAdd.length] = elem.value;
+  });
+
+  if (validData(valuesToAdd)) {
+    isValid = true;
+    valuesToAdd[4] = '$' + Number(valuesToAdd[4]).toLocaleString('en-us');
+
+    const newRow = document.createElement('tr');
+
+    valuesToAdd.forEach((value) => {
+      const column = document.createElement('td');
+
+      column.textContent = value;
+      newRow.append(column);
+    });
+    tbody.append(newRow);
+  }
```

Comment:
The 'validData' function is called to validate the form inputs. Ensure that the function correctly checks all conditions specified in the task, such as the length of the 'name' and the range for 'age'.